### ISSUE TITLE
Problem with installing compiled geany

### ISSUE THREAD

**Issue by CheckB** [geany/geany#2044]
Ubuntu 18.04.
Package assembled through checkinstall.

```
dpkg: dependency problems prevent configuration of geany:
 geany depends on glib2; however:
  Package glib2 is not installed.
 geany depends on gtk2; however:
  Package gtk2 is not installed.
 geany depends on pango; however:
  Package pango is not installed.

```

build:
```
./autogen.sh --disable-html-docs
./configure --disable-html-docs
checkinstall
```

**Comment by CheckB** [comment]
I have postponed the decision for later and not to create another issue, renamed this one.

**Comment by codebrainz** [comment]
For the future, it's better to create a separate issue for separate topics as to not confuse the [comments archive](https://lists.geany.org/pipermail/github-comments/2019-January/016495.html) and people who read/respond to Github mails, which only show the original issue.

For the new issue, you probably need to install the development packages for those listed in the error message. I'm not in Linux at the moment, but they should end with `-dev`. Installing the GTK+2 development package will probably pull in the others, and IIRC it's called `libgtk2.0-dev`.

**Comment by elextr** [comment]
@CheckB since edits are not disseminated what you have done is confuse anyone watching by email. All replies will seem completely off topic.

If you are no longer interested in an issue close with a polite note.

**Comment by CheckB** [comment]
@codebrainz 

I now have libgtk2.0-dev, libgtk2.0-common, libgtk2.0-bin, libgtk2.0-0, but for some reason it still requires a library ...

**Comment by elextr** [comment]
@CheckB you need to install the -dev versions of all the packages configure asks for.

**Comment by codebrainz** [comment]
@CheckB maybe try without `checkinstall`, just replace the command with `make` and see if it compiles. If it does, you'll know it's an issue with `checkinstall` and can troubleshoot that issue separately. As long as you don't run `sudo make install` it won't mess with your system.

**Comment by CheckB** [comment]
geany compiled, but I already put .deb and .deb itself requests the missing libraries for geany :)

**Comment by codebrainz** [comment]
According to [the `checkinstall` man page](https://manpages.debian.org/jessie/checkinstall/checkinstall.8.en.html) you need to specify which packages are required using the `--requires` option. Probably just specifying the `libgtk2.0-dev` package will be enough to pull in the others.

**Comment by LarsGit223** [comment]
@CheckB: I also develop on Ubuntu. In my private wiki I have got the following note about libs which need to be installed (this might not be a minimal list):
```
sudo apt-get install autoconf
sudo apt-get install intltool
sudo apt-get install libglib2.0-dev
sudo apt-get install gtk+2.0
sudo apt-get install libgtk-3-dev
sudo apt-get install libtool
```
Optionally for some tasks (but not required) the following libs and tools can be installed:
```
# For rst2html
sudo apt-get install python3-docutils

# For Source-Code-Check
sudo apt-get install cppcheck

# For generating the source-sode documentation
sudo apt-get install doxygen doxygen-doc doxygen-gui graphviz 
```

**Comment by CheckB** [comment]
@LarsGit223 
The problem is the same. geany still requires:
`` `
dpkg: dependency configuration of geany:
  geany depends on glib2; however:
   Package glib2 is not installed.
  geany depends on gtk2; however:
   Package gtk2 is not installed.
  geany depends on pango; however:
   Package pango is not installed.
`` `

**Comment by elextr** [comment]
@CheckB what is that message from, please always post the command, not just the output?  

Geany requires a Glib2, but the package name is not "glib2" similarly GTK2.  You can see likely real names of the packages in the post from @LarsGit223.

**Comment by CheckB** [comment]
@elextr 
Compile using checkinstall.
checkinstall builds .deb and installs it. Accordingly, after installation, the message appears...

Although all packages are installed

**Comment by elextr** [comment]
@CheckB I said show __the command you used__, checkinstall has many options, its important to know what you actually ran.

**Comment by CheckB** [comment]
@elextr 
No options. ```checkinstall``` and all

**Comment by eht16** [comment]
What do you want to achieve after all?

Geany packages exist already in Ubuntu.
And if you still want to compile yourself, `apt-get build-dep geany` will install anything for you which is necessary to build Geany yourself.

**Comment by b4n** [comment]
OK, trying it myself made me find the problem: Geany's sources include a `.spec` file that was probably designed to build a package for Fedora.  Unfortunately *checkinstall* is reading this, which contains incorrect informations like dependencies when building on Ubuntu.

You should pass the correct flags for checkinstall, or remove the spec file before running it.
Something like this might possibly do it: `checkinstall --type=debian --pkgversion="1.35" --provides="geany-abi-72, geany-api-239" --requires="libglib2.0-0, libgtk3.0-0, libgdk-pixbuf2.0-0" --spec=geany_checkinstall.spec --exclude='*.la'`.  Not tested though :warning:

**Comment by CheckB** [comment]
@b4n 

>  Geany's sources include a .spec file that was probably designed to build a package for Fedora. 

Thx. It works )

**Comment by codebrainz** [comment]
> if the .spec is a distro specific thingy it probably should be removed, especially if it confuses other tools

It didn't confuse `checkinstall`, a tool for generating several kinds of packages, it confused a _user_ of `checkinstall` who happened to be using it for Debian instead of Fedora packages. That said, if nobody is using or maintaining this .spec file, [opting to maintain their own downstream instead](https://src.fedoraproject.org/cgit/rpms/geany.git/tree/geany.spec), then removing it would make sense.

**Comment by eht16** [comment]
See #2064 for removing the spec file.
I guess I added it in the very early days of Geany development to have it or just for playing with it, can't say for sure but I agree that it is most probably not necessary any longer today as many distros already have their own package files.
